pandas/_libs/lib.pyx

```diff
@@ -2208,7 +2208,7 @@ def map_infer(ndarray arr, object f, bint convert=1):
     return result
 
 
-def to_object_array(rows: list, min_width: int=0):
+def to_object_array(rows, int min_width=0):
```

Comment:
type rows as object

Reply:
Will post the benchmark results when they complete.

Reply:
I've rerun twice overnight and reported significant changes don't overlap between runs. I assume this is just the benchmark being noisy. (Commit hashes changed because I rebased again)

```
       before           after         ratio
     [d43ac979]       [8661bbb6]
     <master>         <from-list>
+        50.7±2ms         93.0±3ms     1.84  frame_methods.Iteration.time_iteritems_indexing
+     6.32±0.02μs         7.65±1μs     1.21  timestamp.TimestampOps.time_normalize(tzutc())
+     11.5±0.03μs         13.3±1μs     1.16  offset.OffestDatetimeArithmetic.time_apply(<BusinessMonthEnd>)
+     2.51±0.01μs       2.88±0.2μs     1.15  timeseries.DatetimeIndex.time_get('dst')
+     2.50±0.01μs       2.85±0.2μs     1.14  timeseries.DatetimeIndex.time_get('tz_naive')
+      11.8±0.2μs         13.4±1μs     1.13  timestamp.TimestampConstruction.time_parse_iso8601_tz
+     2.66±0.01μs       3.00±0.2μs     1.13  timeseries.DatetimeIndex.time_get('repeated')
+     20.8±0.07μs         23.1±2μs     1.11  timestamp.TimestampProperties.time_weekday_name(<DstTzInfo 'Europe/Amsterdam' LMT+0:20:00 STD>, None)
-      8.40±0.3ms       7.48±0.2ms     0.89  indexing.NumericSeriesIndexing.time_getitem_scalar(<class 'pandas.core.indexes.numeric.UInt64Index'>, 'nonunique_monotonic_inc')
-      9.62±0.3ms       8.52±0.3ms     0.89  indexing.NumericSeriesIndexing.time_ix_slice(<class 'pandas.core.indexes.numeric.UInt64Index'>, 'nonunique_monotonic_inc')
-      8.85±0.4ms       7.69±0.1ms     0.87  indexing.NumericSeriesIndexing.time_loc_scalar(<class 'pandas.core.indexes.numeric.UInt64Index'>, 'nonunique_monotonic_inc')
-      8.68±0.8μs      7.48±0.03μs     0.86  indexing.CategoricalIndexIndexing.time_get_loc_scalar('monotonic_incr')
-      1.13±0.2μs          896±6ns     0.79  period.PeriodProperties.time_property('min', 'day')
-      86.8±0.6ms          100±4μs     0.00  multiindex_object.GetLoc.time_large_get_loc

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.

       before           after         ratio
     [d43ac979]       [8661bbb6]
     <master>         <from-list>
+        82.4±1μs          104±1μs     1.26  indexing.NumericSeriesIndexing.time_getitem_scalar(<class 'pandas.core.indexes.numeric.Float64Index'>, 'nonunique_monotonic_inc')
+     7.09±0.06μs       8.21±0.8μs     1.16  indexing.NonNumericSeriesIndexing.time_getitem_scalar('datetime', 'unique_monotonic_inc')
+      18.5±0.3μs         20.6±1μs     1.11  indexing.NonNumericSeriesIndexing.time_get_value('datetime', 'unique_monotonic_inc')
-         803±6ms          714±4ms     0.89  gil.ParallelGroupbyMethods.time_parallel(8, 'mean')

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
```